src/App.jsx

```diff
@@ -14,70 +16,64 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGoods, setSelectedGoods] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const handleSelectGoods = (good = '') => {
+    const toSelect = good === selectedGoods ? '' : good;
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+    setSelectedGoods(toSelect);
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGoods.length
+          ? (
+            <>
+              {selectedGoods} is selected
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={handleSelectGoods}
+              />
+            </>
+          )
+          : 'No goods selected'
+        }
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const selected = selectedGoods === good;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            return (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={`${selected && 'has-background-success-light'}`}
```

Comment:
You can also check classNames library. It's a good solution when you need to add className depend on some condition
https://github.com/JedWatson/classnames

Reply:
I prefer use it when I have multiply conditions or long list of classes. Didn't see sence in it here tbh :D